packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -1731,5 +1759,166 @@ describe("SharedString interval collections", () => {
 				assert.equal(collection.detachIndex(mockIntervalIndex), false);
 			});
 		});
+
+		describe("support querying intervals with endpoint in a range", () => {
```

Comment:
One technique I really like using when applicable to get a lot of test coverage with little effort is implementing a "simple" form of the data structure which should give equivalent results, then confirm that it does give the same results. Here, your "simple" form could simply put all of the added intervals in a list and iterate over them all, checking if the endpoint truly is within the range.

Then you can run a number of tests which generate intervals in some fashion (ex: randomly or exhaustively for some possible space) and verify that the efficient spatial index and your "simple" implementation match.